pwnlib/fmtstr.py

```diff
@@ -708,8 +716,52 @@ def make_payload_dollar(data_offset, atoms, numbwritten=0, countersize=4):
 
         # perform write
         if padding:
-            fmt += "%" + str(padding) + "c"
-        fmt += "%" + str(data_offset + idx) + "$" + SPECIFIER[atom.size]
+            # if the padding is less than 3, it is more convenient to write it : [ len("cc") < len("%2c") ] , this could help save some bytes, if it is 3 it will take the same amout of bytes
+            if padding < 4:
+                    fmt += "c" * padding
```

Comment:
I would do it this way, since on no_dollars=True we also add another pointer to the payload.

```suggestion
            if padding < 4 + context.bytes * no_dollars:
                fmt += "c" * padding
```

And actually this won't work for no_dollars, you need to move it up, like 
```py
if padding < 4 + context.bytes * no_dollars:
    fmt += "c" * padding
else:
    fmt += "%" + str(padding) + "c"
    if no_dollars:
        ...
```
instead of the `if padding:` thing.

Reply:
what do you mean by ``on no_dollars=True we also add another pointer to the payload.``
i am not getting why it would be convenient to 
```py
if padding < 4 + context.bytes * no_dollars:
....
```
like let s assume no_dollars is set
it would be 
```py
if padding < 12 : 
.....
```
but isn t more convient to have ``%12c``, than ``"c" * 12 `` ?

Reply:
ohh i got it, if i use "c" * 2 instead of the "%2c" i ll have to nod pack the "c" * 8

Reply:
What I meant is that it is more convenient to have `%nccccccccc%n...ptr1ptr2` than `%n%8c%n...ptr1ccccccccptr2`.

Reply:
dang, you got me on this one, ill change that, gimme a sec

Reply:
ok i ve done that, have a look when u can.

Reply:
i ve made a quiche test seems working. but have a check just to be sure.